### PR TITLE
Add claim check

### PR DESCRIPTION
**Pull request by SidharthK2**
# Add claim check

- Adds Claim warning before Unlock
- Migrate to wagmi v2

image

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/2551

### CODE REVIEW

File: src/hooks/useBridge.ts [hunk 1]
```diff
@@ -23,52 +24,59 @@ export const useBridge = () => {
...
-    abi: erc20Abi,
-    functionName: 'approve',
-    gas: BigInt(calculateGasBuffer(APPROVE)),
-  })
+  // const { writeAsync: approveIq } = useContractWrite({
+  //   address: config.iqAddress as `0x${string}`,
+  //   abi: erc20Abi,
+  //   functionName: 'approve',
+  //   gas: BigInt(calculateGasBuffer(APPROVE)),
+  // })
```
Comment (Tolu-Mals) [inline src/hooks/useBridge.ts:41]:
```suggestion
```

File: src/hooks/useBridge.ts [hunk 2]
```diff
@@ -23,52 +24,59 @@ export const useBridge = () => {
...
+  //   abi: erc20Abi,
+  //   functionName: 'approve',
+  // })
+
+  // const { writeAsync: mint } = useContractWrite({
+  //   address: config.pMinterAddress as `0x${string}`,
+  //   abi: minterAbi,
+  //   functionName: 'mint',
+  //   gas: BigInt(150e3),
+  // })
```
Comment (Tolu-Mals) [inline src/hooks/useBridge.ts:63]:
```suggestion
```

File: src/hooks/useBridge.ts [hunk 3]
```diff
@@ -23,52 +24,59 @@ export const useBridge = () => {
...
+  const { data: mintHash, writeContractAsync: mint } = useWriteContract()
+
+  // const { writeAsync: burn } = useContractWrite({
+  //   address: config.pMinterAddress as `0x${string}`,
+  //   abi: minterAbi,
+  //   functionName: 'burn',
+  // })
```
Comment (Tolu-Mals) [inline src/hooks/useBridge.ts:70]:
```suggestion

```

File: src/hooks/useBridge.ts [hunk 4]
```diff
@@ -23,52 +24,59 @@ export const useBridge = () => {
...
+  const { data: burnHash, writeContractAsync: burn } = useWriteContract()
+
+  // const { writeAsync: redeem } = useContractWrite({
+  //   address: config.pIqAddress as `0x${string}`,
+  //   abi: ptokenAbi,
+  //   functionName: 'redeem',
+  //   gas: BigInt(1e5),
+  // })
```
Comment (Tolu-Mals) [inline src/hooks/useBridge.ts:78]:
```suggestion
```

File: src/hooks/useGaugeCtrl.ts [hunk 5]
```diff
@@ -60,15 +66,17 @@ export const useGaugeCtrl = (nftFarmAddress = config.nftFarmAddress) => {
...
   })
 
-  const { writeAsync: vote } = useContractWrite({
-    ...contractConfig,
-    functionName: 'vote_for_gauge_weights',
-  })
+  // const { writeAsync: vote } = useContractWrite({
+  //   ...contractConfig,
+  //   functionName: 'vote_for_gauge_weights',
+  // })
```
Comment (Tolu-Mals) [inline src/hooks/useGaugeCtrl.ts:77]:
```suggestion
```

File: src/hooks/useNFTGauge.ts [hunk 6]
```diff
@@ -19,58 +20,74 @@ export const useNFTGauge = () => {
...
+  const { data: getRewardHash, writeContract: getReward } = useWriteContract()
 
-  const { writeAsync: getReward } = useContractWrite({
-    ...contractConfig,
-    functionName: 'getReward',
-  })
+  // const { writeAsync: getReward } = useContractWrite({
+  //   ...contractConfig,
+  //   functionName: 'getReward',
+  // })
```
Comment (Tolu-Mals) [inline src/hooks/useNFTGauge.ts:41]:
```suggestion
```

File: src/hooks/useNFTGauge.ts [hunk 7]
```diff
@@ -19,58 +20,74 @@ export const useNFTGauge = () => {
...
+    useWriteContract()
 
-  const { writeAsync: lockMoreBrainy } = useContractWrite({
-    ...contractConfig,
-    functionName: 'lockAdditional',
-  })
+  // const { writeAsync: lockBrainy } = useContractWrite({
+  //   ...contractConfig,
+  //   functionName: 'stakeLocked',
+  // })
```
Comment (Tolu-Mals) [inline src/hooks/useNFTGauge.ts:49]:
```suggestion
```

File: src/hooks/useNFTGauge.ts [hunk 8]
```diff
@@ -19,58 +20,74 @@ export const useNFTGauge = () => {
...
+    useWriteContract()
 
-  const { writeAsync: unlock } = useContractWrite({
-    ...contractConfig,
-    functionName: 'withdrawLocked',
-  })
+  // const { writeAsync: lockMoreBrainy } = useContractWrite({
+  //   ...contractConfig,
+  //   functionName: 'lockAdditional',
+  // })
```
Comment (Tolu-Mals) [inline src/hooks/useNFTGauge.ts:57]:
```suggestion

```

File: src/hooks/useNFTGauge.ts [hunk 9]
```diff
@@ -19,58 +20,74 @@ export const useNFTGauge = () => {
...
+
+  const { data: increaseStakeTimeHash, writeContractAsync: increaseStakeTime } =
+    useWriteContract()
+
+  // const { writeAsync: increaseStakeTime } = useContractWrite({
+  //   ...contractConfig,
+  //   functionName: 'lockLonger',
+  // })
```
Comment (Tolu-Mals) [inline src/hooks/useNFTGauge.ts:65]:
```suggestion
```

File: src/hooks/useNFTGauge.ts [hunk 10]
```diff
@@ -19,58 +20,74 @@ export const useNFTGauge = () => {
...
+
+  const { data: unlockHash, writeContractAsync: unlock } = useWriteContract()
+
+  // const { writeAsync: unlock } = useContractWrite({
+  //   ...contractConfig,
+  //   functionName: 'withdrawLocked',
+  // })
```
Comment (Tolu-Mals) [inline src/hooks/useNFTGauge.ts:72]:
```suggestion
```

**Review by Tolu-Mals — COMMENTED** [review]
There's a bunch of commented out code we need to remove

**Comment by Simplypheyie96** [comment]
Based on the issue, the primary action isn't checkpointing, but rather ensuring that users claim their rewards before proceeding with withdrawals. We have two potential approaches: either users dismiss the modal to take the appropriate action of claiming rewards within the UI, or we provide them with the option to claim rewards directly from the modal itself.

To further improve the UX and encourage users to checkpoint regularly, we could implement a modal in the UI that appears immediately after they lock IQ. This modal would serve as a reminder for users to checkpoint regularly, reinforcing the importance of claiming rewards before unlocking.

@Softdev1

**Comment by Simplypheyie96** [comment]
options we can use for the warning.
image

**Comment by Tolu-Mals** [comment]
@Simplypheyie96 seems like this issue is urgent, we might have to do UI fixes in another pr https://discord.com/channels/920369137828655144/1233026313850458184/1235313552990928957

**Comment by Simplypheyie96** [comment]
> @Simplypheyie96 seems like this issue is urgent, we might have to do UI fixes in another pr https://discord.com/channels/920369137828655144/1233026313850458184/1235313552990928957

i think it's a ux fix, not just ui.

**Review by Tolu-Mals — APPROVED** [review]

**Pull request merged**